RISC-V: Enhancement for stack size to 16-bytes aligned at least

Teng-Shih-Wei opened the pull request:
If enable CONFIG_PMP_STACK_GUARD, the stack size may not 16-bytes
aligned. Enhance this issue.

Signed-off-by: Shih-Wei Teng <swteng@andestech.com>

Teng-Shih-Wei committed:
RISC-V: Enhancement for stack size to 16-bytes aligned at least

If enable CONFIG_PMP_STACK_GUARD, the stack size may not 16-bytes
aligned. Enhance this issue.

Signed-off-by: Shih-Wei Teng <swteng@andestech.com>

lindemer reviewed (changes requested):
@Teng-Shih-Wei Thanks for noticing this and submitting a patch. Would you mind also updating the corresponding Kconfig description? Otherwise, this is OK for me. https://github.com/zephyrproject-rtos/zephyr/blob/2d215484cc352152e9686477fead841c13ebb249/arch/riscv/core/pmp/Kconfig#L30 This config flag has units in bytes but the description is referring to words, which is confusing.

Teng-Shih-Wei committed:
arch: riscv: Update the description of CONFIG_PMP_STACK_GUARD_MIN_SIZE

Update the uints in bytes instead of words in its description. It can
avoid confusion.

Signed-off-by: Shih-Wei Teng <swteng@andestech.com>

Teng-Shih-Wei commented:
@lindemer I update the description. Please help check if it is fine. Thanks!
